src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.CompletionListUpdater.cs

```diff
@@ -376,7 +429,36 @@ private void AddCompletionItems(List<MatchResult> list, CancellationToken cancel
             private VSCompletionItem GetCorrespondingVsCompletionItem(MatchResult matchResult, CancellationToken cancellationToken)
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                return _snapshotData.InitialSortedItemList[matchResult.IndexInOriginalSortedOrder];
+
+                if (matchResult.IndexInOriginalSortedOrder >= 0)
+                    return _snapshotData.InitialSortedItemList[matchResult.IndexInOriginalSortedOrder];
+
+                // This is a "promoted" item added by us, so there's no corresponding VS item from initial sorted list.
+                // We need to craft one based on the data of the original item.
+                var item = matchResult.CompletionItem;
+                if (Helpers.TryGetOriginalIndexOfPromotedItem(item, out var unpromotedIndex))
```

Comment:
why is there not a corresponding vs completion item?  Didn't we change an item that was originally in the list?

Reply:
When we promote a regular item (if it matches one of the defaults and not already a starred item), it's done by creating and adding a duplicate with an added star as prefix (this is consistent with how actual IntelliCode works: we have two otherwise identical items in the list). Therefore the original sorted list passed in from Editor doesn't contain those new items